From 6569a4068bdd2c6df8918971f270d72f2468201b Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Tue, 5 Apr 2022 15:13:38 +0200
Subject: [PATCH] Map the 'verbose' configure option to CMake

Raise the CMake log level to STATUS when the 'verbose' argument is
passed to the configure script.

Change-Id: I736d95ab66b115f8416eec7f9e2ee96d1580c78d
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
---
 cmake/QtProcessConfigureArgs.cmake | 2 ++
 cmake/configure-cmake-mapping.md   | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/QtProcessConfigureArgs.cmake b/cmake/QtProcessConfigureArgs.cmake
index 09bc914f80..493febe077 100644
--- a/cmake/QtProcessConfigureArgs.cmake
+++ b/cmake/QtProcessConfigureArgs.cmake
@@ -120,6 +120,8 @@ while(NOT "${configure_args}" STREQUAL "")
         set(cmake_file_api TRUE)
     elseif(arg STREQUAL "-no-cmake-file-api")
         set(cmake_file_api FALSE)
+    elseif(arg STREQUAL "-verbose")
+        list(APPEND cmake_args "--log-level=STATUS")
     elseif(arg STREQUAL "--")
         # Everything after this argument will be passed to CMake verbatim.
         list(APPEND cmake_args "${configure_args}")
diff --git a/cmake/configure-cmake-mapping.md b/cmake/configure-cmake-mapping.md
index ce90be5e95..1282c0d484 100644
--- a/cmake/configure-cmake-mapping.md
+++ b/cmake/configure-cmake-mapping.md
@@ -9,7 +9,7 @@ The following table describes the mapping of configure options to CMake argument
 | -bindir <dir>                         | -DINSTALL_BINDIR=<dir>                            | similar for -headerdir -libdir and so on                        |
 | -hostdatadir <dir>                    | -DINSTALL_MKSPECSDIR=<dir>                        |                                                                 |
 | -help                                 | n/a                                               | Handled by configure[.bat].                                     |
-| -verbose                              |                                                   |                                                                 |
+| -verbose                              | --log-level=STATUS                                | Sets the CMake log level to STATUS. The default one is NOTICE.  |
 | -continue                             |                                                   |                                                                 |
 | -redo                                 | n/a                                               | Handled by configure[.bat].                                     |
 | -recheck [test,...]                   |                                                   |                                                                 |
-- 
2.26.1.windows.1

